tests/backup/backup_helper.go

```diff
@@ -1629,6 +1667,310 @@ func restoreSuccessWithReplacePolicy(restoreName string, orgID string, retryDura
+func TransformAppContextWithMappings(appContext *scheduler.Context, namespaceMapping map[string]string, storageClassMapping map[string]string, forRestore bool) (*scheduler.Context, error) {
+	appContextNamespace := appContext.ScheduleOptions.Namespace
+	log.Infof("TransformAppContextWithMappings of appContext [%s] with namespace mapping [%v] and storage Class Mapping [%v]", appContextNamespace, namespaceMapping, storageClassMapping)
+
+	restoreAppContext := *appContext
+	var errors []error
+
+	// TODO: remove workaround in future.
+	allStorageClassMappingsPresent := true
+
+	specObjects := make([]interface{}, 0)
+	for _, appSpecOrig := range appContext.App.SpecList {
+		if forRestore {
+			// if we can transforming to obtain a restored specs, VolumeSnapshot should be ignored
+			if obj, ok := appSpecOrig.(*snapv1.VolumeSnapshot); ok {
+				log.Infof("TransformAppContextWithMappings is for restore contexts, ignoring transformation of 'VolumeSnapshot' [%s] in appContext [%s]", obj.Metadata.Name, appContextNamespace)
+				continue
+			} else if obj, ok := appSpecOrig.(*storageapi.StorageClass); ok {
+				log.Infof("TransformAppContextWithMappings is for restore contexts, ignoring transformation of 'StorageClass' [%s] in appContext [%s]", obj.Name, appContextNamespace)
+				continue
+			}
+		}
+
+		appSpec, err := CloneSpec(appSpecOrig) //clone spec to create "restore" specs
+		if err != nil {
+			err := fmt.Errorf("failed to clone spec: '%v'. Err: %v", appSpecOrig, err)
+			errors = append(errors, err)
+			continue
+		}
+		err = TransformToRestoredSpec(appSpec, storageClassMapping)
+		if err != nil {
+			err := fmt.Errorf("failed to TransformToRestoredSpec for %v, with sc map %s. Err: %v", appSpec, storageClassMapping, err)
+			errors = append(errors, err)
+			continue
+		}
+		err = UpdateNamespace(appSpec, namespaceMapping)
+		if err != nil {
+			err := fmt.Errorf("failed to Update the namespace for %v, with ns map %s. Err: %v", appSpec, namespaceMapping, err)
+			errors = append(errors, err)
+			continue
+		}
+		specObjects = append(specObjects, appSpec)
+
+		// TODO: remove workaround in future.
+		if specObj, ok := appSpecOrig.(*corev1.PersistentVolumeClaim); ok {
+			if _, ok := storageClassMapping[*specObj.Spec.StorageClassName]; !ok {
+				allStorageClassMappingsPresent = false
+			}
+		}
+	}
+
+	errStrings := make([]string, 0)
+	for _, err := range errors {
+		if err != nil {
+			errStrings = append(errStrings, err.Error())
+		}
+	}
+
+	if len(errStrings) > 0 {
+		return nil, fmt.Errorf("TransformAppContextWithMappings Errors: {%s}", strings.Join(errStrings, "}\n{"))
+	}
+
+	app := *appContext.App
+	app.SpecList = specObjects
+	restoreAppContext.App = &app
+
+	// we're having to do this as we're under the assumption that `ScheduleOptions.Namespace` will always contain the namespace of the scheduled app
```

Comment:
grammar: we have* to

Reply:
It's a grammatically correct sentence, but I changed it to make more sense.